### PR TITLE
[`BLIP`] Fix daily CI failing test

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

This PR fixes: https://github.com/huggingface/transformers/actions/runs/3754402958/jobs/6378634199

## Why this fix is relevant? 

The reference logits for this test were obtained under pytorch==1.13.1+cu116 and the daily CI uses pytorch==1.13.0+cu116. Setting the tolerance slightly higher (`4e-2`) fixes the test to make it cross-versions compatible. 

cc @LysandreJik @sgugger @ydshieh

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
That is a very big tolerance. It would be better to identify the layer in the model causing this problem.

**Comment by younesbelkada** [comment]
Hmm at the beginning I thought that the `Softmax` was causing the issue, leading to large round errors but the test pass locally with `torch+cu116==1.13.0` but does not pass on the docker image. Will investigate more!

File: tests/models/blip/test_modeling_blip.py [hunk 1]
```diff
@@ -853,7 +853,7 @@ def test_inference_itm(self):
         out_itm = model(**inputs)
         out = model(**inputs, use_itm_head=False)
 
-        expected_scores = torch.Tensor([[0.9779, 0.0221]])
+        expected_scores = torch.Tensor([1.898925542831421])
 
-        self.assertTrue(torch.allclose(torch.nn.Softmax()(out_itm[0].cpu()), expected_scores, atol=1e-3, rtol=1e-3))
-        self.assertTrue(torch.allclose(out[0].cpu(), torch.Tensor([[0.5053]]), atol=1e-3, rtol=1e-3))
+        self.assertTrue(torch.allclose(out_itm[0][0][0].cpu(), expected_scores))
```
Comment (ydshieh) [inline tests/models/blip/test_modeling_blip.py:858]: It would be great if we can figure out why the previous test logic failed between the environment.
Let me know if I could help here, @younesbelkada :-)

**Comment by ydshieh** [comment]
On GCP (my onw / CI runners), all torch versions give

(torch 1.13.x)
```python
[[ 1.9452102 -1.9377887]]
```
or (torch 1.12.1)
```
[[ 1.9452097 -1.9377884]]
```


so
```python
[ 1.9452, -1.9378]
```
will work.

**Comment by ydshieh** [comment]
On GCP (my onw / CI runners), all torch versions give

(torch 1.13.x)
```python
[[0.97982633 0.02017363]]
[[0.50528485]]
```
or (torch 1.12.1)
```
[[0.97982633 0.02017365]]
[[0.5052849]]
```

so
```python
[[0.9798, 0.0202]]
[[0.5053]]
```
will work. Not sure why you got larger differ though, but it is likely an env issue.

**Comment by younesbelkada** [comment]
Thanks so much @ydshieh 💯 , the tests seem to pass now on the CI docker image with your suggested values!

**Review by ydshieh — APPROVED** [review]
Nice 💯 and thank you!

**Review by sgugger — APPROVED** [review]
